src/hooks/useWithdrawals.ts

```diff
@@ -1,27 +1,119 @@
+import { useCelo } from '@celo/react-celo';
 import { useCallback, useEffect, useState } from 'react';
 import { useContracts } from 'src/hooks/useContracts';
+import api from 'src/services/api';
 import { PendingWithdrawal } from 'src/types/account';
-import { Celo } from 'src/types/units';
+import { CeloWei } from 'src/types/units';
+
+export const useWithdrawalBot = (address: string | null) => {
+  const { managerContract, accountContract } = useContracts();
+
+  const finalizeWithdrawal = useCallback(async () => {
+    if (!address) return;
+    const [activeGroups, deprecatedGroups] = await Promise.all([
+      managerContract.methods.getGroups().call(),
+      managerContract.methods.getDeprecatedGroups().call(),
+    ]);
+    const groups = [...activeGroups, ...deprecatedGroups];
+    for (const group of groups) {
+      const scheduledWithdrawals = await accountContract.methods
+        .scheduledWithdrawalsForGroupAndBeneficiary(group, address)
+        .call();
+      if (scheduledWithdrawals === '0') continue;
```

Comment:
Do we need continue here? What do you think about using it?

Reply:
In general I'm not against `continue` or `break` in loops as long as they don't remind goto statements (i.e. when used without labels) and are used as guards. 
However in this case I was able to simplify it, removing `continue` in the process.